x/evm/handler_test.go

```diff
@@ -390,12 +386,12 @@ func (suite *EvmTestSuite) TestOutOfGasWhenDeployContract() {
 	err := tx.Sign(suite.ethSigner, suite.signer)
 	suite.Require().NoError(err)
 
-	snapshotCommitStateDBJson, err := json.Marshal(suite.app.EvmKeeper.CommitStateDB)
+	snapshotCommitStateDBJson, err := json.Marshal(suite.app.EvmKeeper)
 	suite.Require().Nil(err)
 
 	defer func() {
 		if r := recover(); r != nil {
-			currentCommitStateDBJson, err := json.Marshal(suite.app.EvmKeeper.CommitStateDB)
+			currentCommitStateDBJson, err := json.Marshal(suite.app.EvmKeeper)
 			suite.Require().Nil(err)
 			suite.Require().Equal(snapshotCommitStateDBJson, currentCommitStateDBJson)
 		} else {
```

Comment:
we need to remove the json tests, we should not be using JSON to compare state. Instead we should be able to access the Keeper and compare the state.